#!/usr/bin/env python

from dynamic_reconfigure.parameter_generator_catkin import *

PACKAGE_NAME = "squirrel_navigation"

gen = ParameterGenerator()

## PID controller gains.
gen.add("global_frame_id", str_t, 0, "Frame ID of the map.", "/map");
gen.add("kP_lin", double_t, 0, "Proportional gain for linear components", 1.5, 0.0, 20.0)
gen.add("kP_ang", double_t, 0, "Proportional gain for angular component", 1.5, 0.0, 20.0)
gen.add("kI_lin", double_t, 0, "Integral gain for linear components", 0.001, 0.0, 20.0)
gen.add("kI_ang", double_t, 0, "Integral gain for angular component", 0.001, 0.0, 20.0)
gen.add("kD_lin", double_t, 0, "Differential gain for linear components", 0.001, 0.0, 20.0)
gen.add("kD_ang", double_t, 0, "Differential gain for angular component", 0.001, 0.0, 20.0)
gen.add("visualize_topics", bool_t, 0, "", True)

exit(gen.generate(PACKAGE_NAME, "squirrel_navigation", "ControllerPID"))